addons/website/static/src/interactions/video/background_video.js

```diff
@@ -0,0 +1,116 @@
+import { uniqueId } from "@web/core/utils/functions";
+import { throttleForAnimation } from "@web/core/utils/timing";
+import { renderToElement } from "@web/core/utils/render";
+
+class BackgroundVideo extends Interaction {
+    static selector = ".o_background_video";
+    dynamicSelectors = {
+        ...this.dynamicSelectors,
+        _dropdown: () => this.el.closest(".dropdown-menu").parentElement,
+        _modal: () => this.el.closest("modal"),
+    }
+    dynamicContent = {
+        _document: {
+            "t-on-optionalCookiesAccepted": this.onCookieAccepted,
+        },
+        _window: {
+            "t-on-resize": this.throttledForAnimation(() => this.adjustIframe()),
+        },
+        _modal: {
+            "t-on-show.bs.modal": this.onShowModal,
+            "t-on-shown.bs.modal": this.onShownModal,
+        },
+        _dropdown: {
+            "t-on-shown.bs.dropdown": this.throttledForAnimation(() => this.adjustIframe()),
+        },
+    }
+
+    setup() {
+        this.videoSrc = this.el.dataset.bgVideoSrc;
+        this.iframeID = uniqueId("o_bg_video_iframe_");
+    }
+
+    start() {
+        const promise = this.services.youtube_video.setupAutoplay(iframeEl.getAttribute("src"));
+        if (promise) {
+            this.videoSrc += "&enablejsapi=1";
+            promise.then(() => this.appendBgVideo());
+        }
+    }
+
+    adjustIframe() {
+        if (!this.iframe) {
+            return;
+        }
+
+        this.iframe.classList.remove("show");
+
+        // Adjust the iframe
+        var wrapperWidth = this.el.innerWidth;
+        var wrapperHeight = this.el.innerHeight;
+        var relativeRatio = (wrapperWidth / wrapperHeight) / (16 / 9);
+
+        if (relativeRatio >= 1.0) {
+            this.iframe.style.width = "100%";
+            this.iframe.style.height = (relativeRatio * 100) + "%";
+            this.iframe.style.insetInlineStart = "0";
+            this.iframe.style.insetBlockStart = (-(relativeRatio - 1.0) / 2 * 100) + "%";
+        } else {
+            this.iframe.style.width = ((1 / relativeRatio) * 100) + "%";
+            this.iframe.style.height = "100%";
+            this.iframe.style.insetInlineStart = (-((1 / relativeRatio) - 1.0) / 2 * 100) + "%";
+            this.iframe.style.insetBlockStart = "0";
+        }
+
+        void this.iframe.offsetWidth; // Force style addition
+        this.iframe.classList.add("show");
+    }
+
+    appendBgVideo() {
+        const allowedCookies = !this.el.dataset.needCookiesApproval;
+
+        const oldContainer = this.bgVideoContainer || this.querySelector(":scope > .o_bg_video_container");
+        this.bgVideoContainer = renderToElement("website.background.video", {
+            videoSrc: allowedCookies ? this.videoSrc : "about:blank",
+            iframeID: this.iframeID,
+        });
+
+        this.iframe = this.bgVideoContainer.querySelector(".o_bg_video_iframe");
+        this.iframe.addEventListener("load", () => {
```

Comment:
are you sure that you dont mean to use `addListener`

Reply:
I switch it to `addListener` but I don't think it is necessary : The container of the iframe is removed with the interaction.